vllm/utils.py

```diff
@@ -2253,3 +2253,48 @@ def import_pynvml():
     """
     import vllm.third_party.pynvml as pynvml
     return pynvml
+
+
+def warn_for_unimplemented_methods():
+    """
+    A decorator that checks if all abstract methods from the base class 
+    are implemented in the subclass and gives warnings for unimplemented 
+    methods.
+    """
+
+    def decorator(cls: Type[T]) -> Type[T]:
```

Comment:
I think you can also remove this `decorator` function, and let the `warn_for_unimplemented_methods` directly accept the `cls` argument.